crates/router_derive/src/macros/operation.rs

```diff
@@ -231,103 +195,279 @@ impl Conversion {
+                operations = value;
+            }
+            OperationsEnumMeta::Flow { value, .. } => {
+                flows = value;
+            }
+        }
+    }
+
+    if operations.is_empty() {
+        Err(syn::Error::new(
+            Span::call_site(),
+            "atleast one operation must be specitied",
+        ))?;
+    }
+
+    if flows.is_empty() {
+        Err(syn::Error::new(
+            Span::call_site(),
+            "atleast one flow must be specitied",
+        ))?;
+    }
+
+    Ok(OperationProperties { operations, flows })
+}
+
+impl Parse for Derives {
+    fn parse(input: syn::parse::ParseStream<'_>) -> syn::Result<Self> {
+        let text = input.parse::<syn::LitStr>()?;
+        let value = text.value();
+
+        value.as_str().parse().map_err(|_| {
+            let possible_values = Derives::iter()
+                .map(|variants| variants.to_string())
+                .collect::<Vec<_>>()
+                .join(", ");
+
+            syn::Error::new_spanned(
+                &text,
+                format!(
+                    "Unexpected value for flow: `{value}`. Possible values are `{possible_values}`"
+                ),
+            )
         })
-        .cloned()
-        .ok_or_else(|| {
-            helpers::syn_error(
-                Span::call_site(),
-                "Cannot find attribute 'operation' in the macro",
+    }
+}
+
+impl Parse for Conversion {
+    fn parse(input: syn::parse::ParseStream<'_>) -> syn::Result<Self> {
+        let text = input.parse::<syn::LitStr>()?;
+        let value = text.value();
+
+        value.as_str().parse().map_err(|_| {
+            let possible_values = Conversion::iter()
+                .map(|variants| variants.to_string())
+                .collect::<Vec<_>>()
+                .join(", ");
+
+            syn::Error::new_spanned(
+                &text,
+                format!("Unexpected value for operation: `{value}`. Possible values are `{possible_values}`"),
             )
         })
+    }
+}
+
+fn get_conversions(input: syn::parse::ParseStream<'_>) -> syn::Result<Vec<Conversion>> {
+    let lit_str_list = input.parse::<syn::LitStr>()?;
+    lit_str_list
+        .value()
+        .split(",")
+        .into_iter()
+        .map(Conversion::from_str)
+        .map(|result| {
+            result.map_err(|_| {
+                let possible_values = Conversion::iter()
+                    .map(|variant| variant.to_string())
+                    .collect::<Vec<_>>()
+                    .join(", ");
+
+                syn::Error::new(
+                    Span::call_site(),
+                    format!("Unexpected operation, possible values are {possible_values}"),
+                )
+            })
+        })
+        .collect()
+}
+
+fn get_derives(input: syn::parse::ParseStream<'_>) -> syn::Result<Vec<Derives>> {
+    let lit_str_list = input.parse::<syn::LitStr>()?;
+    lit_str_list
+        .value()
+        .split(",")
```

Comment:
Same here, we can allow for optional commas.

Reply:
I have extracted this logic to a common function.